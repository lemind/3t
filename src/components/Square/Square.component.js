import React from 'react';
import './Square.less';

export class SquareComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: props.value,
    };

  }

  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.value !== prevState.value) {
      return {
        value: nextProps.value
      }
    }
    return null
  }

  render() {
    return (
      <div className="square">
        <button
          className="square__button"
          onClick={() => this.props.onClick()}
        >
          { Number.isInteger(this.state.value)
            ? ''
            : this.state.value }
        </button>
      </div>
    )
  };
}
