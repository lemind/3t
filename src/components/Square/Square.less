.square {
  &__button {
    background: #fff;
    border: 1px solid #999;
    float: left;
    font-size: 35px;
    font-weight: bold;
    line-height: 45px;
    height: 45px;
    margin-right: -1px;
    margin-top: -1px;
    padding: 0;
    text-align: center;
    width: 45px;

    &:focus {
      outline: none;
    }
  }
}