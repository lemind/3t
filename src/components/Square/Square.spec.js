import React from 'react';
import { shallow } from 'enzyme';
import { SquareComponent } from './Square.component';

describe('Square', () => {

  beforeEach(() => {
  })

  test('should render SquareComponent', () => {
    expect(shallow(<SquareComponent />)).toMatchSnapshot();
  })
})
