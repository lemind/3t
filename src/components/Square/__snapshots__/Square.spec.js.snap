// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should render SquareComponent 1`] = `
<div
  className="square"
>
  <button
    className="square__button"
    onClick={[Function]}
  />
</div>
`;
